lib/calculator/lexer.rb

```diff
@@ -176,8 +176,255 @@ def do_right_parenthesis
 
     # The primary state of the lexer.  It is not recognising any numeric
     # literals at this point.
+    class State
+      # Handles a whitespace input
+      def do_whitespace
+        [:error]
+      end
+
+      # Handles the end of input
+      def do_end_of_input
+        [:error]
+      end
+
+      # Handles a digit
+      def do_digit(d)
+        [:error]
+      end
+
+      # Handles <tt>-</tt>
+      def do_hyphen
+        [:error]
+      end
+
+      # Handles <tt>.</tt>
+      def do_full_stop
+        [:error]
+      end
+
+      # Handles <tt>+</tt>
+      def do_plus
+        [:error]
+      end
+
+      # Handles <tt>*</tt>
+      def do_asterisk
+        [:error]
+      end
+
+      # Handles <tt>/</tt>
+      def do_slash
+        [:error]
+      end
+
+      # Handles <tt>^</tt>
+      def do_caret
+        [:error]
+      end
+
+      # Handles <tt>(</tt>
+      def do_left_parenthesis
+        [:error]
+      end
+
+      # Handles <tt>)</tt>
+      def do_right_parenthesis
+        [:error]
+      end
+    end
+
+
     class DefaultState < State
       include Singleton
+      def do_whitespace
+        [self]
+      end
+
+      def do_end_of_input
+        [:accept]
+      end
+
+      def do_plus
+        [self, AddOpToken.instance]
+      end
+
+      def do_hyphen
+        [DashState.new]
+      end
+
+      def do_asterisk
+        [self, MultiplyOpToken.instance]
+      end
+
+      def do_slash
+        [self, DivideOpToken.instance]
+      end
+
+      def do_caret
+        [self, ExponentOpToken.instance]
+      end
+
+      def do_left_parenthesis
+        [self, LeftParenthesisToken.instance]
+      end
+
+      def do_right_parenthesis
+        [self, RightParenthesisToken.instance]
+      end
+
+      def do_digit(d)
+        [IntegerState.new(d)]
+        end
+
+    class IntegerState < State
+
```

Comment:
You are missing a constructor here.  As a result, if you tried to run the tests, you'd get an exception telling you that you are passing the wrong number of arguments to IntegerState.new